src/internal/connector/exchange/service_restore.go

```diff
@@ -548,16 +553,22 @@ func CreateContainerDestination(
 			caches[category] = ecc
 			newCache = true
 			directoryCache = ecc
+		} else if did := directoryCache.DestinationNameToID(dest); len(did) > 0 {
+			// calendars are cached by ID, not name, so we need to look up
+			// the destination id before falling back to destination name
+			dest = did
 		}
 
+		folders := append([]string{dest}, directory.Folders()...)
```

Comment:
can you add a quick comment here to say that this update to the path is so that it's resolved properly in the container resolver please. Otherwise it looks a little weird by itself since we're technically mixing IDs and display names (in theory)